projects/benqi-ignite/utils.spec.ts

```diff
@@ -0,0 +1,66 @@
+import { describe, expect, it, vi } from 'vitest';
+import { parseAmount, parseRange, parseRegister, parseWallet } from './utils';
+
+vi.mock('@heyanon/sdk');
+
+type UtilsParams<T extends (...args: any) => any> = Parameters<T>[0] & { expected: ReturnType<T> };
+
+describe('utils', () => {
+    it.each`
+        account                                         | chainName             | expected
+        ${''}                                           | ${'Avalanche'}        | ${{ success: false, errorMessage: 'Wallet not connected' }}
+        ${'not-an-address'}                             | ${'Avalanche'}        | ${{ success: false, errorMessage: 'Expected account to be a valid address' }}
+        ${'0x1234567890123456789012345678901234567890'} | ${'UnsupportedChain'} | ${{ success: false, errorMessage: 'Unsupported chain name: UnsupportedChain' }}
+        ${'0x1234567890123456789012345678901234567890'} | ${'Ethereum'}         | ${{ success: false, errorMessage: 'Protocol is not supported on Ethereum' }}
```

Comment:
👍 

But wondering why you have it in `sdk.ts` ?

Reply:
That a pattern for auto mocking functionality of `vitest`. Notice the path of this file -  `__mocks__/@heyanon/sdk.ts` - this basically triggers vitest to mock `@heyanon/sdk` package in tests, when we call `vi.mock('@heyanon/sdk');` without providing an actual mock implementation